sdk/helper/testhelpers/namespaces/namespaces.go

```diff
@@ -0,0 +1,162 @@
+package namespaces
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// RootNamespacePath is the path of the root namespace.
+const RootNamespacePath = ""
+
+// RootNamespaceID is the ID of the root namespace.
+const RootNamespaceID = "root"
+
+// ErrNotFound is returned by funcs in this package when something isn't found,
+// instead of returning (nil, nil).
+var ErrNotFound = errors.New("not found")
```

Comment:
ErrNotFound, folderPath, and joinPath probably belong elsewhere.  Hopefully I'll figure out where before I promote the PR out of draft state.